stellar-dotnet-sdk/WebAuthentication.cs

```diff
@@ -149,5 +245,32 @@ private static bool ValidateTimeBounds(TimeBounds timeBounds, DateTimeOffset now
             var unixNow = now.ToUnixTimeSeconds();
             return timeBounds.MinTime <= unixNow && unixNow <= timeBounds.MaxTime;
         }
+
+        private static ICollection<string> VerifyTransactionSignatures(Transaction transaction,
+            IEnumerable<string> signers, Network network)
+        {
+            var txHash = transaction.Hash(network);
+            var signaturesUsed = new Dictionary<xdr.DecoratedSignature, string>();
+
+            foreach (var signer in signers)
+            {
+                var keypair = KeyPair.FromAccountId(signer);
+                foreach (var signature in transaction.Signatures)
+                {
+                    if (signaturesUsed.ContainsKey(signature))
+                        continue;
+
+                    if (!signature.Hint.InnerValue.SequenceEqual(keypair.SignatureHint.InnerValue))
+                        continue;
+
+                    if (keypair.Verify(txHash, signature.Signature))
+                    {
+                        signaturesUsed[signature] = keypair.Address;
```

Comment:
It might be a good idea to `break` from the foreach here, since you wouldn't want a signer in the outer foreach to be matched to multiple signatures. To ensure this function never returns an array that has a signer multiple times it might also be a good idea to use a set collection that deduplicates on insert. Instead of using a `Dictionary` above, you could use two sets, one for `signaturesUsed` and one for signers found.

Reply:
I wrote this code with the assumption that every signer can be associated with only one signature (or with the same signature multiple times). In this case, we would perform the assignment `signaturesUsed[signature] = keypair.Address;` multiple times, but it would always be the same `signature` and `keypair.Address`, otherwise the hint wouldn't match or the signature would not be verified. Do you have any example I can use to figure out the issue?

Reply:
Waiting for resolution on this to approve.

Reply:
I don't have a test case that proves otherwise but when verifying signatures it can be worthwhile to make as few assumptions as possible. Example of how we implemented this in the Go SDK, noting that in Go there is no `Set` types and therefore we use two `map`s as sets:

https://github.com/stellar/go/blob/6c651ebf3c2929ed89d74f3eb3684f3ed47cf0e1/txnbuild/transaction.go#L679-L702